### PR TITLE
feat(notifier): reuse pending subscription

### PR DESCRIPTION
**Pull request by vjgene**
resolves https://rsklabs.atlassian.net/browse/RMKT-765

### CODE REVIEW

File: src/api/rif-notifier-service/subscriptionService.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import { logNotImplemented } from 'utils/utils'
+import { NotifierAPIService, NotifierResponse } from './interfaces'
+import { NOTIFIER_RESPONSE_STATUSES } from './models/response'
+import { SubscriptionDTO } from './models/subscriptions'
+
+type SubscriptionResponse = NotifierResponse<Array<SubscriptionDTO>>
+
+export type SubscriptionSummary = Pick<SubscriptionDTO, 'hash' | 'signature' | 'status'>
+
+export const address = 'getSubscriptions' as const
```
Comment (itofarina) [inline src/api/rif-notifier-service/subscriptionService.ts:11]: Might be better to rename this file to `getSubscriptionsService` as subscriptionService is a bit generic?

File: src/api/rif-notifier-service/subscriptionService.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  getActiveSubscription =
+  async (user: string):
+  Promise<SubscriptionDTO|undefined> => {
+    const userSubscriptions: Array<SubscriptionDTO> = await this.getUserSubscriptions(user)
+    return this._getActiveSubscription(userSubscriptions)
+  }
+
+  _getActiveSubscription = (userSubscriptions: Array<SubscriptionDTO>): SubscriptionDTO|undefined => {
```
Comment (itofarina) [inline src/api/rif-notifier-service/subscriptionService.ts]:
```suggestion
  _getActiveSubscriptionFrom = (userSubscriptions: Array<SubscriptionDTO>): SubscriptionDTO|undefined => {
```
to differentiate a bit from the other function

**Review by itofarina — APPROVED** [review]
LGTM - just optional suggestions

**Review by jurajpiar — APPROVED** [review]

File: src/api/rif-notifier-service/subscriptionService.ts (hunk 1 above)
Comment (vjgene) [reply src/api/rif-notifier-service/subscriptionService.ts:11]: I believe getSubscriptionService is not a good name for the class or file. May be we can call it SubscriptionReadService or SubscriptionFetcher or something

**Pull request merged**